test/tests/performance/pbs_node_buckets_perf.py

```diff
@@ -65,13 +65,43 @@ def cust_attr_func(self, name, totalnodes, numnode, attribs):
         """
         a = {'resources_available.color': self.colors[numnode % 7]}
         return dict(attribs.items() + a.items())
+    
+    def submit_jobs(self, attribs, num):
+        """
+        Submit num jobs each in their individual equiv class
+        """
+        wt = 100
+        jids = []
+        for i in range(num):
+            attribs['Resource_List.walltime'] = wt + i
+            J = Job(TEST_USER, attrs=attribs)
+            jid = self.server.submit(J)
+            jids.append(jid)
+            
+        return jids
+    
+    def run_cycle(self):
+        """
+        Run a cycle and return the length of the cycle
+        """
+        t = int(time.time())
```

Comment:
If the scheduler was already in a scheduling cycle when you executed this then there could be a race condition in this function. I think we should first check that server/scheduler state is idle before doing anything in this function.

Reply:
Good point.  When I added the test, we were in cycle.  I hadn't turned scheduling off before submitting all 1000 jobs.  I did that and checked for server_stat = Idle.